File: SUPPORTING FILES/SPCHOL304/Completed/CS/Ex1/BDCEx1/BDCEx1/BdcModel1/CustomerService.cs
//--------------------------------------------------------------------------------
// This file is a "Sample" from the SharePoint Foundation 2010
// Samples
//
// Copyright (c) Microsoft Corporation. All rights reserved.
//
// This source code is intended only as a supplement to Microsoft
// Development Tools and/or on-line documentation.  See these other
// materials for detailed information regarding Microsoft code samples.
// 
// THIS CODE AND INFORMATION ARE PROVIDED AS IS WITHOUT WARRANTY OF ANY
// KIND, EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE
// IMPLIED WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A
// PARTICULAR PURPOSE.
//--------------------------------------------------------------------------------

using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace BDCEx1.BdcModel1
{
    /// <summary>
    /// All the methods for retrieving, updating and deleting data are implemented in this class file.
    /// The samples below show the finder and specific finder method for Entity1.
    /// </summary>
    public class CustomerService
    {
        public static Dictionary<Int32, Customer> d = null;

        public static Customer ReadItem(Int32 id)
        {
            // take a copy for SharePoint
            Customer c = new Customer();
            Customer e = d[id];
            c.CustomerId = e.CustomerId;
            c.FirstName = e.FirstName;
            c.LastName = e.LastName;
            c.Message = e.Message;
            return c;
        }

        public static IEnumerable<Customer> ReadList()
        {
            // this is usually the first method called so check for null
            if (d == null)
            {
                d = new Dictionary<Int32, Customer>();
                for (int i = 0; i < 10; i++)
                {
                    Customer e = new Customer();
                    e.CustomerId = i;
                    e.Message = i + " Item Data";
                    e.FirstName = i + " First Name";
                    e.LastName = i + " Last Name";
                    d.Add(i, e);
                }
            }
            return d.Values;
        }

        public static void Update(Customer customer, Int32 id)
        {
            d[id].FirstName = customer.FirstName;
            d[id].LastName = customer.LastName;
            d[id].Message = customer.Message;
        }

    }
}

File: SUPPORTING FILES/SPCHOL304/Completed/VB/Ex1/BDCEx1/BDCEx1/BdcModel1/CustomerService.vb
'--------------------------------------------------------------------------------
' This file is a "Sample" from the SharePoint Foundation 2010
' Samples
'
' Copyright (c) Microsoft Corporation. All rights reserved.
'
' This source code is intended only as a supplement to Microsoft
' Development Tools and/or on-line documentation.  See these other
' materials for detailed information regarding Microsoft code samples.
' 
' THIS CODE AND INFORMATION ARE PROVIDED AS IS WITHOUT WARRANTY OF ANY
' KIND, EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE
' IMPLIED WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A
' PARTICULAR PURPOSE.
'--------------------------------------------------------------------------------

Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text

''' <summary>
''' All the methods for retrieving, updating and deleting data are implemented in this class file.
''' The samples below show the finder and specific finder method for Entity1.
''' </summary>
''' 

Public Class CustomerService

    Public Shared d As Dictionary(Of Integer, Customer) = Nothing

    Public Shared Function ReadItem(ByVal id As Integer) As Customer
        Dim e As New Customer()
        Dim c = d([id])
        e.CustomerId = c.CustomerId
        e.Message = c.Message
        e.FirstName = c.FirstName
        e.LastName = c.LastName
        Return e
    End Function

    Public Shared Function ReadList() As IEnumerable(Of Customer)

        If d Is Nothing Then
            d = New Dictionary(Of Integer, Customer)
            For i As Integer = 0 To 9
                Dim e As New Customer()
                e.CustomerId = i
                e.Message = e.CustomerId & " Item Data"
                e.FirstName = e.CustomerId & " First Name"
                e.LastName = e.CustomerId & "Last Name"
                d.Add(i, e)
            Next
        End If

        Return d.Values
    End Function

    Public Shared Sub Update(ByVal customer As Customer, ByVal id As Integer)
        d([id]).FirstName = customer.FirstName
        d([id]).LastName = customer.LastName
        d([id]).Message = customer.Message
    End Sub
End Class


